src/main.c

```diff
@@ -194,24 +200,30 @@ void init_task(void *arg)
             &eth_sender_handle);
 
     configASSERT(status);
-
-
-
-    gpio.Mode = GPIO_MODE_OUTPUT_PP;
-    gpio.Pull = GPIO_NOPULL;
-    gpio.Pin = GPIO_PIN_13;
-    HAL_GPIO_Init(GPIOD, &gpio);
-    HAL_GPIO_WritePin(GPIOD, GPIO_PIN_13, GPIO_PIN_SET);
-
-    for (;;) {
-        if (!netif_is_link_up(netif)) {
+    uint8_t reset_state = 1;

```

Comment:
reset_state is used only inside for(;;) cycle below, so why not to declare it as static inside the for { .... } - this will reduce it's scope.